documentation/src/main/asciidoc/public/reference/_search-dsl-predicate.adoc

```diff
@@ -856,6 +856,13 @@ but can be <<search-dsl-predicate-common-constantScore,made constant with `.cons
 * The score of an `and` predicate can be <<search-dsl-predicate-common-boost,boosted>>
 with a call to `.boost(...)`.
 
+[[search-dsl-predicate-and-limitations]]
+=== Limitations
+
+Keep in mind that adding a <<search-dsl-predicate-knn,knn>> as an `and` clause will lead to an exception when using
+<<backend-elasticsearch-compatibility-elasticsearch,an Elastic distribution of Elasticsearch>>.
+See <<search-dsl-predicate-knn-limitations,`knn` limitations>> for more details.
```

Comment:
I think there was a misunderstanding, I meant you should add the limitation to `backend-elasticsearch-compatibility-elasticsearch`.

You certainly can keep this, though, it doesn't hurt.

Reply:
added this paragraph:
https://github.com/hibernate/hibernate-search/blob/764d121153449680246cf7b91f677a3db30982a8/documentation/src/main/asciidoc/public/reference/_backend-elasticsearch.adoc?plain=1#L50-L53